File: AppointmentSchedulerAPI/layers/BusinessLogicLayer/Model/AppointmentNotification.cs
using AppointmentSchedulerAPI.layers.BusinessLogicLayer.Model.Types.Notification;

namespace AppointmentSchedulerAPI.layers.BusinessLogicLayer.Model
{
    public class AppointmentNotification : Notification
    {
        public AppointmentNotification()
        {
            this.Type = NotificationType.APPOINTMENT_NOTIFICATION;
        }

        public required AppointmentNotificationCodeType Code { get; set; }
        public required virtual AppointmentIdentifiers Appointment { get; set; }
    }
}

File: AppointmentSchedulerWeb/src/view-model/business-entities/notification/appointment-notification.ts
import { AppointmentNotificationCodeType } from "../types/appointment-notification-code.types";
import { NotificationStatusType } from "../types/notification-status.types";
import { NotificationType } from "../types/notification.types";
import { NotificationBase } from "./notification-base";

export class AppointmentNotification extends NotificationBase {
  constructor(
    createdAt: Date,
    uuid: string,
    message: string,
    type: NotificationType,
    status: NotificationStatusType,
    public code: AppointmentNotificationCodeType,
    public appointmentUuid: string,
  ) {
    super(createdAt, uuid, message, type, status);
  }

}
